platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1271,14 +1271,35 @@ private void warnOrFailOnInvalidUsage(String methodName, boolean allowDeprecated
         if (!isProperUsage(properUsages)) {
             String currentUsageDesc = UsageDescriber.describeCurrentUsage(this);
             String properUsageDesc = ProperMethodUsage.summarizeProperUsage(properUsages);
-            String msgTemplate = "Calling configuration method '%s' is not allowed for configuration '%s', which has permitted usage(s):\n" +
-                "%s\n" +
-                "This method is only meant to be called on configurations which allow the %susage(s): '%s'.";
 
-            GradleException ex = new GradleException(String.format(msgTemplate, methodName, getName(), currentUsageDesc, allowDeprecated ? "" : "(non-deprecated) ", properUsageDesc));
+            String prefixTemplate = "Calling configuration method '%s' is not allowed for configuration '%s'";
+            String suffixTemplate = "This method is only meant to be called on configurations which allow the %susage(s): '%s'.";
+            GradleException ex = new GradleException(
+                String.format(
+                    prefixTemplate + ", which has permitted usage(s):\n%s\n" + suffixTemplate,
+                    methodName,
+                    getName(),
+                    currentUsageDesc, allowDeprecated ? "" : "(non-deprecated) ", properUsageDesc
+                )
+            );
+
             ProblemId id = ProblemId.create("method-not-allowed", "Method call not allowed", GradleCoreProblemGroup.configurationUsage());
             throw configurationServices.getProblems().getInternalReporter().throwing(ex, id, spec -> {
-                spec.contextualLabel(ex.getMessage());
+                spec.contextualLabel(
+                    String.format(
+                        prefixTemplate,
+                        methodName,
+                        getName()
+                    )
+                );
+                spec.details(
+                    String.format(
+                        "'%s' has the following permitted usage(s):\n%s\n" + suffixTemplate,
+                        getName(),
+                        currentUsageDesc,
+                        allowDeprecated ? "" : "(non-deprecated) ", properUsageDesc
```

Comment:
💄 I think this is more readable, even if taller
```suggestion
                        allowDeprecated ? "" : "(non-deprecated) ",
                        properUsageDesc
```